sympy/stats/drv.py

```diff
@@ -69,6 +69,21 @@ def cdf(self, x, **kwargs):
         """ Cumulative density function """
         return self.compute_cdf(**kwargs)(x)
 
+    @cacheit
+    def compute_characteristic_function(self, **kwargs):
+        """ Compute the characteristic function from the PDF
+
+        Returns a Lambda
+        """
+        x, t = symbols('x, t', integer=True, finite=True, cls=Dummy)
```

Comment:
t is declared to be integer here but it should be real. Having it integer may lead to incorrect simplification of the sum, e.g., if the argument of `exp` ends up being an integer multiple of 2*pi*I for integer t.